packages/app/components/Queue/Student/StudentBanner.tsx

```diff
@@ -54,7 +54,7 @@ export default function StudentBanner({
   leaveQueue,
 }: StudentBannerProps): ReactElement {
   const { studentQuestion, studentQuestionIndex } = useStudentQuestion(queueId);
-  const isQueueOnline = useQueue(queueId).queue?.room === "Online";
+  const isQueueOnline = useQueue(queueId).queue?.room.startsWith("Online");
```

Comment:
I did a full text search of the codebase for usages of the `teams.microsoft` link and it looks like there are a few places where we forgot to put in the `isOnline` check in the first place 💀 🤦 
* [when we open a call for question grouping](https://github.com/sandboxnu/office-hours/blob/58dbb1e1b92a58453add7930d1122596c4030364/packages/app/components/Queue/TA/QuestionGrouping/AllQuestionsChecklist.tsx#L106)
* [the 'help next' button on the queue page](https://github.com/sandboxnu/office-hours/blob/954592349d63c09e1a7696e8b4883434b749b254/packages/app/components/Queue/TA/TAQueue.tsx#L108)
   * It looks like I also forgot to swap the default teams message here to use the `useDefaultMessage` hook, if you could also do that it would be awesome!!

It kind of feels like we've run into issues with forgetting to update all the places that open a teams call several times now, so I'm wondering if it would be worth making some sort of util to abstract that logic.. you could maybe do a function that given some information like the users and default msg, creates the teams link and calls `window.open` with it? Or idk maybe a custom hook that returns a function to open the window?

Reply:
Aye aye captain

Reply:
oh wait i have a question: am i applying the `isOnline` check in those places you mentioned in the `disabled` field or conditionally opening the window with an `if`?

for example in `AllQuestionsChecklist.tsx`, this snippet is in `BannerPrimaryButton`:
```
      window.open(
        `https://teams.microsoft.com/l/chat/0/0?users=${usersInLink}&message=${defaultMessage}`
      );
    }}
    disabled={!canHelp || checkedQuestions.size === 0}
```

cause if it's the former, i think it would be hard to abstract. also, the two files that i already refactored use both ways repsectively

Reply:
Hm I'm not sure I 100% understand what you're saying, but the idea is to just conditionally do `window.open` based on if the queue is online. The `disabled` is separate from that - eg. you might not be able to help someone because you're already helping someone else, which is unrelated to the queue being online. 

Conditionally calling window.open also may or may not be in that helper you abstract out, depending on what you decide to go with 🤷

Reply:
I understand the assignment now; thanks chief